test/functional/page_objects/settings_page.ts

```diff
@@ -486,6 +486,46 @@ export function SettingsPageProvider({ getService, getPageObjects }: FtrProvider
       await this.clickSaveScriptedField();
     }
 
+    async addRuntimeField(name: string, type: string, script: string) {
+      await this.clickAddField();
+      await this.setFieldName(name);
+      await this.setFieldType(type);
+      if (script) {
+        await this.setFieldScript(script);
+      }
+      await this.clickSaveField();
+      await new Promise((resolve) => setTimeout(resolve, 1000 * 30));
+    }
+
+    async clickAddField() {
+      log.debug('click Add Field');
+      await testSubjects.click('addField');
+    }
+
+    async clickSaveField() {
+      log.debug('click Save');
+      await testSubjects.click('fieldSaveButton');
+    }
+
+    async setFieldName(name: string) {
+      log.debug('set field name = ' + name);
+      await testSubjects.setValue('nameField', name);
+    }
+
+    async setFieldType(type: string) {
+      log.debug('set type = ' + type);
+      await testSubjects.setValue('typeField', type);
+    }
+
+    async setFieldScript(script: string) {
+      log.debug('set script = ' + script);
+      await (await testSubjects.findAll('formRowToggle'))[1].click();
+      await new Promise((resolve) => setTimeout(resolve, 1000 * 5));
```

Comment:
Working with the monaco script editor presents some unique challenges. I'm uncertain how to programmatically determine when its ready and loaded

Reply:
We definitely want to gravitate to `retry.waitFor()` calls to wait for a specific condition instead of using `setTimeout()`. There's an example at `x-pack/test/accessibility/apps/ingest_node_pipelines.ts` line 24 that can be used as a reference.